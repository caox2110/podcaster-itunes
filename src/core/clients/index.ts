export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
