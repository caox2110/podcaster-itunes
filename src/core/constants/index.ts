import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
