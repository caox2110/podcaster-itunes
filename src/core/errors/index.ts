export { default as ExceptionHandler } from './Exceptions.handler';
