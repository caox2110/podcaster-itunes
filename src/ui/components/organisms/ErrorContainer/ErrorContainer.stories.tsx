import type { ComponentStory, ComponentMeta } from '@storybook/react';
import { withPerformance } from 'storybook-addon-performance';

import ErrorContainer from '.';

const Template: ComponentStory<typeof ErrorContainer> = (properties) => (
  <ErrorContainer {...properties} />
);

const ErrorContainerWithMessage = Template.bind({});
ErrorContainerWithMessage.args = {
  error: 'Error fetching podcast-rss. Message: Internal Server Error.',
};

export { ErrorContainerWithMessage };
export default {
  title: 'UI/Organisms/ErrorContainer',
  component: ErrorContainer,
  decorators: [withPerformance],
  argTypes: {
    error: {
      description: 'Error message to show in the component.',
      type: { name: 'string' },
      defaultValue: '',
      control: 'text',
    },
  },
} as ComponentMeta<typeof ErrorContainer>;
