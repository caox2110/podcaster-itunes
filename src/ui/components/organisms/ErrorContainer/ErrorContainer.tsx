import { ReactElement, ReactNode } from 'react';

import { MuiBox, MuiContainer, MuiTypography } from '../..';
import { styled } from '../../../themes';

const ContentStyle = styled('div')(({ theme }) => ({
  maxWidth: 480,
  margin: 'auto',
  minHeight: '60vh',
  display: 'flex',
  justifyContent: 'center',
  flexDirection: 'column',
  padding: theme.spacing(12, 0),
}));

export default function ErrorContainer({
  error = 'App Error',
}: {
  error: string | ReactElement | ReactNode;
}) {
  return (
    <MuiContainer>
      <ContentStyle sx={{ textAlign: 'center', alignItems: 'center' }}>
        <MuiTypography paragraph variant='h3'>
          Sorry, there was an error:
        </MuiTypography>
        <pre style={{ whiteSpace: 'normal' }}>
          <MuiTypography sx={{ color: 'text.secondary' }}>{error}</MuiTypography>
        </pre>
        <MuiBox
          alt='app error image'
          component='img'
          src='/static/illustrations/undraw_server_down_s-4-lk.svg'
          sx={{ height: { xs: 200, sm: 260 }, mx: 'auto', my: { xs: 5, sm: 10 } }}
        />
      </ContentStyle>
    </MuiContainer>
  );
}
