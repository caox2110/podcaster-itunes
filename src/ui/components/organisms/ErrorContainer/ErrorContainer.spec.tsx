import { ReactElement, ReactNode } from 'react';

import { render, screen } from '@/core/test';
import type { RenderResult } from '@/core/test';

import ErrorContainer from '.';

const properties = {
  error: 'App Error',
};

function renderApp({
  ...rest
}: {
  error: string | ReactElement | ReactNode;
} = properties): RenderResult {
  const utils = render(<ErrorContainer {...rest} />);
  return {
    ...utils,
  };
}

describe('ErrorContainer', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('Should render a children without throwing any errors', () => {
      renderApp();

      expect(screen.getByText(/app error/i)).toBeInTheDocument();
    });
  });
});
