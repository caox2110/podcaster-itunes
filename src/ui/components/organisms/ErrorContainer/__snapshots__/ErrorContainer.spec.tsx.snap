// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContainer Rendering Should render without throwing any errors 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="css-1rhfahi"
    >
      <p
        class="MuiTypography-root MuiTypography-h3 MuiTypography-paragraph css-1rp5ucn-MuiTypography-root"
      >
        Sorry, there was an error:
      </p>
      <pre
        style="white-space: normal;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-7lj4m2-MuiTypography-root"
        >
          App Error
        </p>
      </pre>
      <img
        alt="app error image"
        class="MuiBox-root css-zkplgh"
        src="/static/illustrations/undraw_server_down_s-4-lk.svg"
      />
    </div>
  </div>
</div>
`;
